### PR TITLE
Nightly test for vMotion-VCH-Appliance

### PR DESCRIPTION
**Pull request by singhshwetaQE**
Fixes # vmware/vic-ui#419

### CODE REVIEW

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    Log To Console   Finished Creating vMotion Setup
+    
+Gather All vSphere Logs
+    ${hostList}=  Run  govc ls -t HostSystem host/cls | xargs
+    Run  govc logs.download ${hostList}
+
+*** Test Cases ***
+Test
+   Log To Console  Deploy VIC to the VC cluster...
+   Deploy OVA And Install UI Plugin And Run Regression Tests  13-01-vMotion  vic-*.ova  %{TEST_DATASTORE}  %{BRIDGE_NETWORK}  %{PUBLIC_NETWORK}  %{TEST_USERNAME}  %{TEST_PASSWORD}
```
Comment (mdharamadas1) [inline tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: test name needs update

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+   Log To Console  vMotion VCH...
+   Set Environment Variable  VCH-NAME  ${VCH}
```
Comment (mdharamadas1) [inline tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: where is `${VCH}` set?

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+   ${host}=  Get VCH Host Name  ${VCH-IP}
+
+   ${status}=  Run Keyword And Return Status  Should Contain  ${host}  ${esx1-ip}
```
Comment (mdharamadas1) [inline tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: where is `${esx1-ip}` set?

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+   Run Keyword If  ${status}  Run  govc vm.migrate -host cls/${esx2-ip} %{VCH-NAME}
```
Comment (mdharamadas1) [inline tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: These steps are only migrating VCH, instead I think this test should verify migrating OVA appliance? Because VCH migration is already covered in `vic` repo I guess?

File: tests/resources/VC-Util.robot [hunk 5]
```diff
@@ -67,6 +67,13 @@ Get VM Host Name
     ${host}=  Fetch From Right  @{out}[-1]  ${SPACE}
     [Return]  ${host}
 
+Get VCH Host Name
```
Comment (mdharamadas1) [inline tests/resources/VC-Util.robot]: Actually this is getting a VM host name, so call it as `Get VM Name` may be? And correct the parameter to be more specific from `${vm}` to `${vm-ip}`

File: tests/resources/page-objects/Vsphere-VCH-Plugin-Util.robot [hunk 6]
```diff
@@ -201,5 +201,6 @@ Create VCH using UI And Set Docker Parameters
     Click Finish Button
     Unselect Frame
     Wait Until Page Does Not Contain  VCH name
+    Set Global Variable   ${VCH}  ${name}
```
Comment (mdharamadas1) [inline tests/resources/page-objects/Vsphere-VCH-Plugin-Util.robot]: In other keywords, we have been using `Set Test Variable` to avoid concurrency issues if any later, so lets use that and to be clear, i would suggest renaming variable `${VCH}` to `${VCH-NAME}`

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot (hunk 3 above)
Comment (singhshwetaQE) [reply tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: Gather Host IPs

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot (hunk 1 above)
Comment (singhshwetaQE) [reply tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: will fix that

File: tests/resources/VC-Util.robot (hunk 5 above)
Comment (singhshwetaQE) [reply tests/resources/VC-Util.robot]: It's getting VCH host ip, so I will change into Get VCH Host IP

Comment (singhshwetaQE) [reply tests/resources/VC-Util.robot]: will change into ${vm-ip}

File: tests/resources/page-objects/Vsphere-VCH-Plugin-Util.robot (hunk 6 above)
Comment (singhshwetaQE) [reply tests/resources/page-objects/Vsphere-VCH-Plugin-Util.robot]: I will change it Set Test Variable and ${VCH-NAME}

File: tests/resources/VC-Util.robot (hunk 5 above)
Comment (mdharamadas1) [reply tests/resources/VC-Util.robot]: Ok sorry confused, is it getting IP or host name?
Anyways, Its part of the VC util and the fact that its getting VM info in general, I would rename it to `Get VM...`. If it is named like `Get VCH..` then it wouldn't make sense to use it for any other VM.

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot [hunk 7]
```diff
@@ -0,0 +1,83 @@
+# Copyright 2016-2017 VMware, Inc. All Rights Reserved.
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: 2018

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+    Log To Console  Set environment variables up for GOVC
+    Set Environment Variable  GOVC_INSECURE  1
+    Set Environment Variable  GOVC_URL  ${vc-ip}
+    Set Environment Variable  GOVC_USERNAME  Administrator@vsphere.local
+    Set Environment Variable  GOVC_PASSWORD  Admin\!23
+
+    Add Host To Distributed Switch  /vcqaDC/host/cls
+
+    Log To Console  Enable DRS and VSAN on the cluster
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: The below is not enabling vsan

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    Set Environment Variable  PUBLIC_NETWORK  vm-network
+    Remove Environment Variable  TEST_DATACENTER
+    Set Environment Variable  TEST_DATASTORE  vsanDatastore
+    Set Environment Variable  TEST_RESOURCE  /vcqaDC/host/cls
+    Set Environment Variable  TEST_TIMEOUT  15m
+
+    Gather Host IPs
+    Log To Console   Finished Creating vMotion Setup
+
+Gather All vSphere Logs
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: This needs to be in a util, not here

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    ${hostList}=  Run  govc ls -t HostSystem host/cls | xargs
+    Run  govc logs.download ${hostList}
+
+*** Test Cases ***
+Test
+   Log To Console  Deploy VIC to the VC cluster...
+   Deploy OVA And Install UI Plugin And Run Regression Tests  5-16-vMotion  vic-*.ova  %{TEST_DATASTORE}  %{BRIDGE_NETWORK}  %{PUBLIC_NETWORK}  %{TEST_USERNAME}  %{TEST_PASSWORD}
+
+   Log To Console  vMotion VCH...
+   Set Environment Variable  VCH-NAME  ${VCH-NAME}
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: Not really clear why you are doing this here, just need to use the suite var form below instead of switching to env var use

File: tests/resources/VC-Util.robot [hunk 11]
```diff
@@ -67,6 +67,13 @@ Get VM Host Name
     ${host}=  Fetch From Right  @{out}[-1]  ${SPACE}
     [Return]  ${host}
 
+Get VM Host IP
```
Comment (mhagen-vmware) [inline tests/resources/VC-Util.robot]: Get VM host by ip

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+Resource  ../../resources/Util.robot
+Suite Setup    Wait Until Keyword Succeeds  10x  10m  vMotion Setup
+Suite Teardown  Run Keyword And Ignore Error  Nimbus Cleanup  ${list}
+Test Teardown  Run Keyword If Test Failed  Gather All vSphere Logs
+
+*** Keywords ***
+vMotion Setup
+    [Timeout]    110 minutes
+    Run Keyword And Ignore Error  Nimbus Cleanup  ${list}  ${false}
+    ${name}=  Evaluate  'vic-vmotion-' + str(random.randint(1000,9999))  modules=random
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: Add 5-16 to the name here so we are a bit more specific

File: tests/resources/page-objects/Vsphere-VCH-Plugin-Util.robot [hunk 13]
```diff
@@ -201,5 +201,6 @@ Create VCH using UI And Set Docker Parameters
     Click Finish Button
     Unselect Frame
     Wait Until Page Does Not Contain  VCH name
+    Set Test Variable   ${VCH-NAME}  ${name}
```
Comment (mhagen-vmware) [inline tests/resources/page-objects/Vsphere-VCH-Plugin-Util.robot]: Unclear why this change is being made in this pr

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot (hunk 4 above)
Comment (singhshwetaQE) [reply tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: @mdharamadas1  I will be creating separate test for OVA vMotion, checked with Mike.

File: tests/resources/page-objects/Vsphere-VCH-Plugin-Util.robot (hunk 13 above)
Comment (singhshwetaQE) [reply tests/resources/page-objects/Vsphere-VCH-Plugin-Util.robot]: I can't access ${name} in my test

File: tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot (hunk 10 above)
Comment (singhshwetaQE) [reply tests/manual-test-cases/Group5-Interoperability-Tests/5-16-vMotion-VCH-Appliance.robot]: will fix this

File: tests/resources/VC-Util.robot (hunk 11 above)
Comment (singhshwetaQE) [reply tests/resources/VC-Util.robot]: will fix this
